base/dataset.go

```diff
@@ -219,7 +201,7 @@ func UnpinDataset(r repo.Repo, ref repo.DatasetRef) error {
 // * dsp.BodyPath being a url
 // * dsp.BodyPath being a path on the local filesystem
 // TODO - consider moving this func to some other package. maybe actions?
-func DatasetPodBodyFile(dsp *dataset.DatasetPod) (cafs.File, error) {
+func DatasetPodBodyFile(store cafs.Filestore, dsp *dataset.DatasetPod) (cafs.File, error) {
```

Comment:
With all the `else`s used in this function, there is something that seems off about this function.

Perhaps, to be more clear,  we bail if `BodyPath` is empty, before we start checking all the different `BodyPath` cases? That way they can be all if statements that return at the end of each statement.